modules/Bio/EnsEMBL/Hive/DBSQL/AnalysisJobAdaptor.pm

```diff
@@ -587,6 +587,54 @@ sub reset_or_grab_job_by_dbID {
     return $job;
 }
 
+=head2 reset_job_by_input_id_and_sync
+
+  Arg [1]: string $input_id pattern
+  Arg [2]: array $analysis_id
+  Example:
+    my $job = $self->{'dba'}->get_AnalysisJobAdaptor->reset_job_by_input_id_and_sync($input_id_pattern, $analyses_pattern);
+  Description:
+    Reset jobs for the specified $input_id pattern and analyses_id pattern. $input_id can be a wildcard argument.
+  Returntype : none
+  Exceptions :
+  Caller     : beekeeper.pl
+
+=cut
+
+sub reset_job_by_input_id_and_sync {
+    my ($self, $input_id_pattern, @analyses_list) = @_;
+    $input_id_pattern =~ s/\%/\.*/g;
```

Comment:
The pattern you currently accept is in fact any Perl regexp, with the addition of substituting `%` with `.*` to give it a MySQL taste. However, `a%b` means "starts with `a` and ends with `b`" in MySQL, but "contains an `a` and then a `b` in Perl. Is that fine ? @ens-bwalts ?

Reply:
To be consistent with other pattern matching (such as -analyses_pattern in runWorker.pl), this should be MySql style - so a%b should be starts with a and ends with b, and if the user wants to match any leading string, it would need to be %a%b

Reply:
:+1: